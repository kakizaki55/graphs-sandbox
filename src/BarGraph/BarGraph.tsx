import React from 'react'
import { BarChart, XAxis, YAxis, CartesianGrid, Tooltip, Legend, Bar } from 'recharts';

interface BarGraphProps {
  data?: any,
  size?: any
}


const BarGraph = ({ data, size }:BarGraphProps) => {
  return (
  <>
    <div>Bargraph</div>
    <BarChart width={size.width} height={size.height} data={data}>
      <CartesianGrid strokeDasharray="3 3" />
      <XAxis dataKey="name" />
      <YAxis />
      <Tooltip />
      <Legend />
      <Bar dataKey="Success Rate" fill="#8884d8" />
      <Bar dataKey="distance from home traveled" fill="#82ca9d" />
      <Bar dataKey="Total run time" fill="#8f22dd" />
    </BarChart>
  </>
  )}

  export default BarGraph
