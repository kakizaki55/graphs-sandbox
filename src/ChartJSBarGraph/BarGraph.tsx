import React from 'react'
// import 'chart.js/auto';
import { Bar } from 'react-chartjs-2';

interface BarGraphProps {
    data: any
    size?: any
}

const styles={
  backGroudColor: '#32a0a8'
}


const BarGraph = ({data , size={width: 1000, height: 500}}: BarGraphProps) => {

  
  return (  
      <div >
        <div>Chart.js</div>
          <div style={{ 
              width: size.width, 
              height: size.height 
              }}>
            <Bar data={{...data}} ></Bar>
          </div>
    </div>
  )
}

export default BarGraph
