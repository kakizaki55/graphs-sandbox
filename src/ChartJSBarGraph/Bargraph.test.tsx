import React from "react";
import '@testing-library/jest-dom'
import {  render, screen } from "@testing-library/react"
import  BarGraph from "./BarGraph";
import robotsDataFormatted from "./data"
import 'jest-canvas-mock';
// @ts-nocheck
window.ResizeObserver =
window.ResizeObserver ||
jest.fn().mockImplementation(() => ({
    disconnect: jest.fn(),
    observe: jest.fn(),
    unobserve: jest.fn(),
}));

describe('BarGraph', () => {
    describe('testing to make sure barGraph displays the correct values', () => {
        it('should sure barGraph displays the correct values', () => {
            
            const parser = new DOMParser()

            const size = {width: 1000, height: 500}
            

            const { container } = render(<BarGraph data={robotsDataFormatted} size={size} />)
            
            const canvasWrapper: HTMLCanvasElement = screen.getByRole('img')
            // console.log('canvasWrapper', canvasWrapper.getContext("2d").__getEvents())
            // console.log('datasetr', parser.parseFromString(canvasWrapper.dataset.toString(),"text/html"))
            // const ctx = canvasWrapper.getContext()
            const context = canvasWrapper.getContext('2d')
            console.log('S1', context.getContextAttributes())

            expect(canvasWrapper).toMatchSnapshot()
            expect(true).toBe(true)
            
            // eslint-disable-next-line testing-library/no-debugging-utils
            screen.debug()
        })
    })
})