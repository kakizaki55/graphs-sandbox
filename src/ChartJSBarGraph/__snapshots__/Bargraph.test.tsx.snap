// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarGraph testing to make sure barGraph displays the correct values should sure barGraph displays the correct values 1`] = `
<canvas
  height="0"
  role="img"
  style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
  width="0"
/>
`;
